crates/kubelet/src/server.rs

```diff
@@ -68,15 +68,47 @@ fn get_ping() -> Response<Body> {
 /// Implements the kubelet path /containerLogs/{namespace}/{pod}/{container}
 async fn get_container_logs<T: Provider + Sync>(
     provider: &T,
-    _req: &Request<Body>,
+    req: &Request<Body>,
 ) -> Response<Body> {
-    // TODO: extract the right data from the request.
+    // Basic validation steps
+    let path: Vec<&str> = req.uri().path().split('/').collect();
+    // Because of the leading slash, index 0 is an empty string. Index 1 is the
+    // container logs path
+    let (namespace, pod, container) = match path.as_slice() {
+        [_, _, namespace, pod, container] => (*namespace, *pod, *container),
```

Comment:
Maybe add: `!if namespace. is_empty() & ...` to this? You can then get rid of the other check below

Reply:
I think for now I'll leave it as I am knee deep in another task.